sling.go

```diff
@@ -432,11 +437,43 @@ func (s *Sling) Receive(successV, failureV interface{}) (*http.Response, error)
 // the response is returned.
 // Receive is shorthand for calling Request and Do.
 func (s *Sling) ReceiveWithContext(ctx context.Context, successV, failureV interface{}) (*http.Response, error) {
-	req, err := s.RequestWithContext(ctx)
+	req, err := s.requestWithContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return s.doDecode(req, successV, failureV)
+}
+
+func (s *Sling) Do(ctx context.Context) (*http.Response, error) {
```

Comment:
I don't think this makes sense. I think you should have to pass in the request. It should look just like the `http` package signature IMHO

Reply:
Talked about this in person and decided to leave it as is.